### ISSUE TITLE
`@safe` setter for driver

### ISSUE THREAD

**Issue by MartinNowak** [MartinNowak/io#15]
Track number of open files/sockets in driver to allow `@safe` setting/swapping of the IO Driver.

**Comment by schveiguy** [comment]
You may be able to migrate the open files as well, if you pull them out of the current driver as file descriptors, and open them in the new driver.

**Comment by MartinNowak** [comment]
That meant keeping track of all open files within the driver, not a reasonable option. Also driver changes don't seem to have a use-case.
Furthermore an async driver would mark sockets as non-blocking during opening, migrating such flags seems unfeasible either.

**Comment by schveiguy** [comment]
I actually meant manually, not automatically . I’m thinking of the standard handles which you don’t want to close.

**Comment by schveiguy** [comment]
I agree that changing drivers mid-program doesn't really have a great use case. However, it should be supported on entry to `main`.

**Comment by anon17** [comment]
btw, you can store shared driver in TLS:
```d
struct LocalShared(T)
{
	shared T value;
	alias value this;
}
LocalShared!Driver _driver;
```
